src/Collection.php

```diff
@@ -554,6 +619,31 @@ public function dropIndexes(array $options = [])
         return $operation->execute($server);
     }
 
+    /**
+     * Drop a single Atlas Search index in the collection.
+     * Only available when used against a 7.0+ Atlas cluster.
+     *
+     * @param string                 $name    Search index name
+     * @param array{comment?: mixed} $options Additional options
+     * @throws UnsupportedException if options are not supported by the selected server
+     * @throws InvalidArgumentException for parameter/option parsing errors
+     * @throws DriverRuntimeException for other driver errors (e.g. connection errors)
+     */
+    public function dropSearchIndex(string $name, array $options = []): void
+    {
+        $operation = new DropSearchIndex($this->databaseName, $this->collectionName, $name);
+        $server = select_server($this->manager, $options);
+
+        try {
+            $operation->execute($server);
+        } catch (CommandException $e) {
+            // Suppress namespace not found errors for idempotency
+            if ($e->getCode() !== 26) {
```

Comment:
A spec e2e test expect this error to be ignored. The error code `26` comes from [mongo source](https://github.com/mongodb/mongo/blob/master/src/mongo/base/error_codes.yml#L64)

Reply:
[NamespaceNotFound Errors](https://github.com/mongodb/specifications/blob/78554de6a78fc996fa8d1ef9733b123a29ad8309/source/index-management/index-management.rst#namespacenotfound-errors) spec

Reply:
I think this should be handled in the operation class. See DropCollection and its `ERROR_CODE_NAMESPACE_NOT_FOUND` constant for prior art.

The benefit of doing so is that other libraries that use the operation directly (admittedly a hypothetical) can also benefit from the behavior. I took this a step further with the CSFLE collection management operation classes introduced earlier this year.